tests/522_explicit_struct_names.rs

```diff
@@ -0,0 +1,43 @@
+use ron::{Options, extensions::Extensions, Error, from_str};
+use serde_derive::Deserialize;
+
+#[derive(Debug, Deserialize)]
+struct Id(u32);
+
+#[derive(Debug, Deserialize)]
+struct Foo {
+    #[allow(unused)]
+    pub id: Id,
+}
+
+const EXPECT_ERROR_MESSAGE: &'static str = "expected `Err(Error::ExpectedStructName)`, deserializer returned `Ok`";
+const INCORRECT_ERROR_MESSAGE: &'static str = "expected error ExpectedStructName, found";
+
+#[test]
+fn explicit_struct_names() {
+    let options = Options::default()
+        .with_default_extension(Extensions::EXPLICIT_STRUCT_NAMES);
+
+    // phase 1
+    let content = r#"(
+        id: Id(3),
+    )"#;
+    let foo = options.from_str::<Foo>(content);
+    match foo.expect_err(EXPECT_ERROR_MESSAGE).code {
+        Error::ExpectedStructName(_) => {},
+        err => panic!("{INCORRECT_ERROR_MESSAGE} \"{err}\""),
+    }
+
+    // phase 2
+    let content = r#"Foo(
+        id: (3),
+    )"#;
+    let foo = options.from_str::<Foo>(content);
+    match foo.expect_err(EXPECT_ERROR_MESSAGE).code {
+        Error::ExpectedStructName(_) => {},
+        err => panic!("{INCORRECT_ERROR_MESSAGE} \"{err}\""),
+    }
+
+    // phase 3 (use content from phase 2)
+    let _foo = from_str::<Foo>(content).unwrap();
```

Comment:
I use three phases: one to check if the extension works with regular `struct`s, one to check if it works with `newtype structs`, and one to ensure no errors occur with the default extensions. Should I add one to check if it works with `tuple structs` as well?

Reply:
Sure, that's a good idea!